src/syntax/Parser.js

```diff
@@ -771,34 +790,58 @@ export class Parser {
 
   parseNamedExport_() {
     // NamedExport ::=
-    //     "*" "from" ModuleSpecifier(load)
-    //     ExportSpecifierSet ("from" ModuleSpecifier(load))?
+    //   "*" "from" ModuleSpecifier
+    //   ExportSpecifierSet
+    //   "*" "from" ModuleSpecifier
+    //   "*" "as" Identifier "from" ModuleSpecifier
+    //   Identifier "from" ModuleSpecifier
     let start = this.getTreeStartLocation_();
+    let exportClause, moduleSpecifier = null;
 
-    let specifierSet, expression = null;
+    switch (this.peekType_()) {
+      case OPEN_CURLY:
+        exportClause = this.parseExportSpecifierSet_();
+        if (this.peekPredefinedString_(FROM)) {
+          this.eatId_(FROM);
+          moduleSpecifier = this.parseModuleSpecifier_();
+        } else {
+          // Ensure that the bindings (lhs) of the specifiers are not keywords.
+          // Keywords are only disallowed when we do not have a 'from' following
+          // the ExportSpecifierSet.
```

Comment:
Well the code says the opposite:
Unless 'from' follows the ExportSpecifierSet, keywords are not allowed.